### PR TITLE
Rework SFX rendering to fix giant sprites.

### PR DESCRIPTION
**Pull request by fluffyfreak**
This was just an attempt to fix the SFX rendering of smoke and explosions (#3681)
In that regard it has some improvements but still needs more work.

Whlist doing that though I realised just how awful the whole SFX rendering is so have rewritten it a bit.

### CODE REVIEW

File: src/Sfx.cpp [hunk 1]
```diff
@@ -107,152 +88,200 @@ void Sfx::TimeStepUpdate(const float timeStep)
...
+void SfxManager::Cleanup()
+{
+	for(size_t t=TYPE_EXPLOSION; t<TYPE_NONE; t++) 
+	{
+		const size_t numInstances = GetNumberInstances(SFX_TYPE(t));
+		if(!numInstances)
+			continue;
+
+		for (Sint64 i = Sint64(numInstances-1); i >= 0 ; i--)
+		{
```
Comment (impaktor) [inline src/Sfx.cpp]: space semicolon?

(Sorry, still in proofreading mindset after thesis handin)

Comment (fluffyfreak) [inline src/Sfx.cpp]: That's ok :)

**Comment by walterar** [comment]
@fluffyfreak  How to build the matrix explosions.png?
Manually? or with some utilitary program, from image0...31.png?

**Comment by fluffyfreak** [comment]
@walterar you can use something like [TexturePacker](https://www.codeandweb.com/texturepacker) or you can do it manually. 
The first frame of the animation is in the top left, the 2nd is too it's right, when you reach the end of that row of 6 images you drop down onto the next row of 6 images.
Currently it's hardcoded too a grid of 6x6, but only the first 32 are used for the explosions.

**Comment by walterar** [comment]
@fluffyfreak  TexturePacker worked well for me, thanks.
I'm using 36 images, right 6x6, with some little changes in sfx.cpp. In static ships it seems to work very well. I need to test with small ships in a "real" battle with 4 or 5 pirates.

**Comment by impaktor** [comment]
Changes to png's worries me due to the growing git repository size, so I opt for causion when changes must be made to png. So is the scout+ version objectivly better, or does it not fit the pioneer style? If the former, maybe we should use that version instead?

Just a though.

**Comment by fluffyfreak** [comment]
@impaktor the current 32 image texture atlas is fine, I won't be changing it unless @nozmajner wants to make new explosion animations in which case I'll make some larger code changes to drive it from configuration files. Right now it isn't necessary as it would just add 4 frames to the animation.

**Comment by impaktor** [comment]
But I assume @walterar uses other animations in Scout+, and now wants to know how to update to the new format you introduce here? (Maybe I should just look at scout+)

**Comment by walterar** [comment]
@fluffyfreak I like how it works; smoke needs a little more love. ;)
Have you thought about the posibility to have (a random or directed) explosions_00..99.png?
explosions.png?

**Comment by fluffyfreak** [comment]
@walterar I'm sure then that #3697 will be of interest to you :)

**Pull request merged**